src/staticanalysis/bot/static_analysis_bot/clang/tidy.py

```diff
@@ -69,7 +69,7 @@ class ClangTidy(object):
     def __init__(self, validate_checks=True):
         self.binary = os.path.join(
             os.environ['MOZBUILD_STATE_PATH'],
-            'clang-tools', 'clang', 'bin', 'clang-tidy',
+            'clang-tools', 'clang-tidy', 'bin', 'clang-tidy',
```

Comment:
Why do we even need the binary? Can't we always use the `mach static-analysis` command? I see it is used only to list the available checks, couldn't we add a mach command to do that, so we don't need to hardcode the clang-tidy path?

(This is not needed for the current PR, but for the future)

Reply:
CC @abpostelnicu

Reply:
We have to download and setup the binary in `MOZBUILD_STATE` before mach, or it triggers the artifact download, [which fails on taskcluster](https://dxr.mozilla.org/mozilla-central/source/python/mozbuild/mozbuild/mach_commands.py#1421)